hipposerve/web/auth.py

```diff
@@ -0,0 +1,115 @@
+"""
+Authentication layer for the Web UI. Provides two core dependencies:
+
+- `LoggedInUser` - a dependency that requires a user to be logged in. If the user is not logged in, a 401 Unauthorized response is returned.
+- `PotentialLoggedInUser` - a dependency that returns a user if they are logged in, or `None` if they are not.
+"""
+
+from datetime import datetime, timedelta, timezone
+from typing import Annotated
+
+import jwt
+from beanie import PydanticObjectId
+from fastapi import Depends, HTTPException, Request, status
+from fastapi.openapi.models import OAuthFlows
+from fastapi.security import OAuth2
+from fastapi.security.utils import get_authorization_scheme_param
+from pydantic import BaseModel
+
+from hipposerve.service import users as users_service
+from hipposerve.settings import SETTINGS
+
+
+class OAuth2PasswordBearerWithCookie(OAuth2):
+    def __init__(
+        self,
+        tokenUrl: str,
+        scheme_name: str = None,
+        scopes: dict = None,
+        auto_error: bool = True,
```

Comment:
I think it would be useful to have a docstring here, useful to understand what are available scheme_names, scopes and what does auto_error do (it seems strange to handle an error with a bool instead of an exception.

Reply:
I've added one; all of this comes directly from the built in FastAPI auth schemes.